### PR TITLE
boards: xtensa: adsp: select IMX for imx8 board

### LINKED ISSUE #38216: nxp_adsp_imx8 fails to build a number of tests
`./scripts/twister -p nxp_adsp_imx8`

Gets:

```
ERROR   - nxp_adsp_imx8             samples/kernel/condition_variables/condvar/sample.kernel.cond_var FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/samples/kernel/condition_variables/condvar/sample.kernel.cond_var/build.log
INFO    - Total complete:   11/  88  12%  skipped: 1316, failed:    1
ERROR   - nxp_adsp_imx8             samples/kernel/condition_variables/simple/sample.kernel.cond_var.simple FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/samples/kernel/condition_variables/simple/sample.kernel.cond_var.simple/build.log
INFO    - Total complete:   12/  88  13%  skipped: 1316, failed:    2
ERROR   - nxp_adsp_imx8             tests/kernel/fatal/message_capture/kernel.logging.message_capture FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/fatal/message_capture/kernel.logging.message_capture/build.log
INFO    - Total complete:   13/  88  14%  skipped: 1316, failed:    3
ERROR   - nxp_adsp_imx8             tests/kernel/sleep/kernel.common.timing            FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sleep/kernel.common.timing/build.log
INFO    - Total complete:   14/  88  15%  skipped: 1316, failed:    4
ERROR   - nxp_adsp_imx8             tests/kernel/condvar/condvar_api/kernel.condvar    FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/condvar/condvar_api/kernel.condvar/build.log
INFO    - Total complete:   15/  88  17%  skipped: 1316, failed:    5
ERROR   - nxp_adsp_imx8             tests/kernel/timer/starve/kernel.timer.starve      FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/timer/starve/kernel.timer.starve/build.log
INFO    - Total complete:   16/  88  18%  skipped: 1316, failed:    6
ERROR   - nxp_adsp_imx8             tests/kernel/fatal/exception/kernel.common.stack_sentinel FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/fatal/exception/kernel.common.stack_sentinel/build.log
INFO    - Total complete:   17/  88  19%  skipped: 1316, failed:    7
ERROR   - nxp_adsp_imx8             tests/kernel/tickless/tickless_concept/kernel.tickless.concept FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/tickless/tickless_concept/kernel.tickless.concept/build.log
INFO    - Total complete:   18/  88  20%  skipped: 1316, failed:    8
ERROR   - nxp_adsp_imx8             tests/kernel/sched/metairq/kernel.scheduler.metairq FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sched/metairq/kernel.scheduler.metairq/build.log
INFO    - Total complete:   19/  88  21%  skipped: 1316, failed:    9
ERROR   - nxp_adsp_imx8             tests/kernel/semaphore/semaphore/kernel.semaphore  FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/semaphore/semaphore/kernel.semaphore/build.log
INFO    - Total complete:   20/  88  22%  skipped: 1316, failed:   10
ERROR   - nxp_adsp_imx8             tests/kernel/workq/critical/kernel.workqueue       FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/workq/critical/kernel.workqueue/build.log
INFO    - Total complete:   21/  88  23%  skipped: 1316, failed:   11
ERROR   - nxp_adsp_imx8             tests/kernel/timer/timer_monotonic/kernel.timer.monotonic FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/timer/timer_monotonic/kernel.timer.monotonic/build.log
INFO    - Total complete:   22/  88  25%  skipped: 1316, failed:   12
ERROR   - nxp_adsp_imx8             tests/kernel/timer/timer_api/kernel.timer          FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/timer/timer_api/kernel.timer/build.log
INFO    - Total complete:   23/  88  26%  skipped: 1316, failed:   13
ERROR   - nxp_adsp_imx8             tests/kernel/workq/work/kernel.work.api            FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/workq/work/kernel.work.api/build.log
INFO    - Total complete:   24/  88  27%  skipped: 1316, failed:   14
ERROR   - nxp_adsp_imx8             tests/kernel/workq/work_queue/kernel.workqueue     FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/workq/work_queue/kernel.workqueue/build.log
INFO    - Total complete:   25/  88  28%  skipped: 1316, failed:   15
ERROR   - nxp_adsp_imx8             tests/kernel/timer/timer_error_case/kernel.timer   FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/timer/timer_error_case/kernel.timer/build.log
INFO    - Total complete:   26/  88  29%  skipped: 1316, failed:   16
ERROR   - nxp_adsp_imx8             tests/kernel/stack/stack/kernel.stack.usage        FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/stack/stack/kernel.stack.usage/build.log
INFO    - Total complete:   27/  88  30%  skipped: 1316, failed:   17
ERROR   - nxp_adsp_imx8             tests/kernel/sched/schedule_api/kernel.scheduler.multiq_no_timeslicing FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sched/schedule_api/kernel.scheduler.multiq_no_timeslicing/build.log
INFO    - Total complete:   28/  88  31%  skipped: 1316, failed:   18
ERROR   - nxp_adsp_imx8             tests/kernel/sched/schedule_api/kernel.scheduler.dumb_no_timeslicing FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sched/schedule_api/kernel.scheduler.dumb_no_timeslicing/build.log
INFO    - Total complete:   29/  88  32%  skipped: 1316, failed:   19
ERROR   - nxp_adsp_imx8             tests/kernel/timer/timer_api/kernel.timer.tickless FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/timer/timer_api/kernel.timer.tickless/build.log
INFO    - Total complete:   30/  88  34%  skipped: 1316, failed:   20
ERROR   - nxp_adsp_imx8             tests/kernel/sched/schedule_api/kernel.scheduler.multiq FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sched/schedule_api/kernel.scheduler.multiq/build.log
INFO    - Total complete:   33/  88  37%  skipped: 1318, failed:   21
ERROR   - nxp_adsp_imx8             tests/kernel/queue/kernel.queue                    FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/queue/kernel.queue/build.log
INFO    - Total complete:   34/  88  38%  skipped: 1318, failed:   22
ERROR   - nxp_adsp_imx8             tests/kernel/sched/schedule_api/kernel.scheduler.no_timeslicing FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sched/schedule_api/kernel.scheduler.no_timeslicing/build.log
INFO    - Total complete:   35/  88  39%  skipped: 1318, failed:   23
ERROR   - nxp_adsp_imx8             tests/kernel/pending/kernel.objects                FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/pending/kernel.objects/build.log
INFO    - Total complete:   37/  88  42%  skipped: 1319, failed:   24
ERROR   - nxp_adsp_imx8             tests/kernel/poll/kernel.poll                      FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/poll/kernel.poll/build.log
INFO    - Total complete:   38/  88  43%  skipped: 1319, failed:   25
ERROR   - nxp_adsp_imx8             tests/kernel/profiling/profiling_api/kernel.common.profiling FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/profiling/profiling_api/kernel.common.profiling/build.log
INFO    - Total complete:   39/  88  44%  skipped: 1319, failed:   26
ERROR   - nxp_adsp_imx8             tests/kernel/sched/deadline/kernel.scheduler.deadline FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sched/deadline/kernel.scheduler.deadline/build.log
INFO    - Total complete:   40/  88  45%  skipped: 1319, failed:   27
ERROR   - nxp_adsp_imx8             tests/kernel/sched/preempt/kernel.scheduler.preempt FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sched/preempt/kernel.scheduler.preempt/build.log
INFO    - Total complete:   41/  88  46%  skipped: 1319, failed:   28
ERROR   - nxp_adsp_imx8             tests/kernel/pipe/pipe/kernel.pipe                 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/pipe/pipe/kernel.pipe/build.log
INFO    - Total complete:   50/  88  56%  skipped: 1327, failed:   29
ERROR   - nxp_adsp_imx8             tests/kernel/pipe/pipe_api/kernel.pipe.api         FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/pipe/pipe_api/kernel.pipe.api/build.log
INFO    - Total complete:   51/  88  57%  skipped: 1327, failed:   30
ERROR   - nxp_adsp_imx8             tests/kernel/sched/schedule_api/kernel.scheduler   FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/sched/schedule_api/kernel.scheduler/build.log
INFO    - Total complete:   55/  88  62%  skipped: 1330, failed:   31
ERROR   - nxp_adsp_imx8             tests/kernel/mutex/sys_mutex/system.mutex.nouser   FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mutex/sys_mutex/system.mutex.nouser/build.log
INFO    - Total complete:   58/  88  65%  skipped: 1331, failed:   32
ERROR   - nxp_adsp_imx8             tests/kernel/msgq/msgq_usage/kernel.message_queue_usage FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/msgq/msgq_usage/kernel.message_queue_usage/build.log
INFO    - Total complete:   59/  88  67%  skipped: 1331, failed:   33
ERROR   - nxp_adsp_imx8             tests/kernel/mutex/mutex_api/kernel.mutex          FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mutex/mutex_api/kernel.mutex/build.log
INFO    - Total complete:   60/  88  68%  skipped: 1331, failed:   34
ERROR   - nxp_adsp_imx8             tests/kernel/mem_slab/mslab_concept/kernel.memory_slabs.concept FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_slab/mslab_concept/kernel.memory_slabs.concept/build.log
INFO    - Total complete:   61/  88  69%  skipped: 1331, failed:   35
ERROR   - nxp_adsp_imx8             tests/kernel/mem_slab/mslab/kernel.memory_slabs.api FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_slab/mslab/kernel.memory_slabs.api/build.log
INFO    - Total complete:   62/  88  70%  skipped: 1331, failed:   36
ERROR   - nxp_adsp_imx8             tests/kernel/msgq/msgq_api/kernel.message_queue    FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/msgq/msgq_api/kernel.message_queue/build.log
INFO    - Total complete:   63/  88  71%  skipped: 1331, failed:   37
ERROR   - nxp_adsp_imx8             tests/kernel/mem_protect/sys_sem/kernel.memory_protection.sys_sem FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_protect/sys_sem/kernel.memory_protection.sys_sem/build.log
INFO    - Total complete:   64/  88  72%  skipped: 1331, failed:   38
ERROR   - nxp_adsp_imx8             tests/kernel/mem_protect/sys_sem/kernel.memory_protection.sys_sem.nouser FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_protect/sys_sem/kernel.memory_protection.sys_sem.nouser/build.log
INFO    - Total complete:   65/  88  73%  skipped: 1331, failed:   39
ERROR   - nxp_adsp_imx8             tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api/build.log
INFO    - Total complete:   66/  88  75%  skipped: 1331, failed:   40
ERROR   - nxp_adsp_imx8             tests/kernel/mem_slab/mslab_threadsafe/kernel.memory_slabs.threadsafe FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_slab/mslab_threadsafe/kernel.memory_slabs.threadsafe/build.log
INFO    - Total complete:   70/  88  79%  skipped: 1334, failed:   41
ERROR   - nxp_adsp_imx8             tests/kernel/mem_protect/stack_random/kernel.memory_protection.stack_random FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_protect/stack_random/kernel.memory_protection.stack_random/build.log
INFO    - Total complete:   71/  88  80%  skipped: 1334, failed:   42
ERROR   - nxp_adsp_imx8             tests/kernel/threads/thread_apis/kernel.threads.apis FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/threads/thread_apis/kernel.threads.apis/build.log
INFO    - Total complete:   72/  88  81%  skipped: 1334, failed:   43
ERROR   - nxp_adsp_imx8             tests/kernel/mem_heap/k_heap_api/kernel.k_heap_api FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_heap/k_heap_api/kernel.k_heap_api/build.log
INFO    - Total complete:   73/  88  82%  skipped: 1334, failed:   44
ERROR   - nxp_adsp_imx8             tests/kernel/mbox/mbox_api/kernel.mailbox.api      FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mbox/mbox_api/kernel.mailbox.api/build.log
INFO    - Total complete:   74/  88  84%  skipped: 1334, failed:   45
ERROR   - nxp_adsp_imx8             tests/kernel/mbox/mbox_usage/kernel.mailbox.usage  FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mbox/mbox_usage/kernel.mailbox.usage/build.log
INFO    - Total complete:   75/  88  85%  skipped: 1334, failed:   46
ERROR   - nxp_adsp_imx8             tests/kernel/lifo/lifo_usage/kernel.lifo.usage     FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/lifo/lifo_usage/kernel.lifo.usage/build.log
INFO    - Total complete:   76/  88  86%  skipped: 1334, failed:   47
ERROR   - nxp_adsp_imx8             tests/kernel/early_sleep/kernel.common.sleep       FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/early_sleep/kernel.common.sleep/build.log
INFO    - Total complete:   77/  88  87%  skipped: 1334, failed:   48
ERROR   - nxp_adsp_imx8             tests/kernel/mem_heap/mheap_api_concept/kernel.memory_heap FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/mem_heap/mheap_api_concept/kernel.memory_heap/build.log
INFO    - Total complete:   78/  88  88%  skipped: 1334, failed:   49
ERROR   - nxp_adsp_imx8             tests/kernel/threads/thread_init/kernel.threads.init FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/threads/thread_init/kernel.threads.init/build.log
INFO    - Total complete:   79/  88  89%  skipped: 1334, failed:   50
ERROR   - nxp_adsp_imx8             tests/kernel/fifo/fifo_usage/kernel.fifo.usage     FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/fifo/fifo_usage/kernel.fifo.usage/build.log
INFO    - Total complete:   80/  88  90%  skipped: 1334, failed:   51
ERROR   - nxp_adsp_imx8             tests/kernel/context/kernel.common                 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/context/kernel.common/build.log
INFO    - Total complete:   81/  88  92%  skipped: 1334, failed:   52
ERROR   - nxp_adsp_imx8             tests/kernel/fifo/fifo_timeout/kernel.fifo.timeout FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/fifo/fifo_timeout/kernel.fifo.timeout/build.log
INFO    - Total complete:   82/  88  93%  skipped: 1334, failed:   53
ERROR   - nxp_adsp_imx8             tests/kernel/lifo/lifo_api/kernel.lifo             FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/lifo/lifo_api/kernel.lifo/build.log
INFO    - Total complete:   83/  88  94%  skipped: 1334, failed:   54
ERROR   - nxp_adsp_imx8             tests/kernel/threads/thread_stack/kernel.threads.thread_stack FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/threads/thread_stack/kernel.threads.thread_stack/build.log
INFO    - Total complete:   84/  88  95%  skipped: 1334, failed:   55
ERROR   - nxp_adsp_imx8             tests/kernel/fifo/fifo_api/kernel.fifo             FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/fifo/fifo_api/kernel.fifo/build.log
INFO    - Total complete:   85/  88  96%  skipped: 1334, failed:   56
ERROR   - nxp_adsp_imx8             tests/kernel/device/kernel.device                  FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/device/kernel.device/build.log
INFO    - Total complete:   86/  88  97%  skipped: 1334, failed:   57
ERROR   - nxp_adsp_imx8             tests/kernel/interrupt/arch.interrupt              FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/interrupt/arch.interrupt/build.log
INFO    - Total complete:   87/  88  98%  skipped: 1334, failed:   58
ERROR   - nxp_adsp_imx8             tests/kernel/device/kernel.device.pm               FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/nxp_adsp_imx8/tests/kernel/device/kernel.device.pm/build.log
```

### PR DESCRIPTION
**Pull request by iuliana-prodan**
For some platforms, like NXP's IMX8 or Mediatek's MT8195,
the size of an interrupt vector table entry is 0x1C bytes,
less than usual (0x30 for Intel's platforms).
So, the interrupt handlers don't fit in the vector table entries.
    
I've added a small indirection to bypass this size
constraint and moved the default handlers to the end
of vector table, renaming them to _Level\LVL\()VectorHelper.
For this, I've added a generic configuration -
XTENSA_SMALL_VECTOR_TABLE_ENTRY and enabled it for
NXP_ADSP_IMX8 board.

Fixes #38216
Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by MaureenHelm — COMMENTED** [review]
This symbol is defined in SOF, shouldn't it be configured there instead?

**Comment by iuliana-prodan** [comment]
> This symbol is defined in SOF, shouldn't it be configured there instead?

It is configured in SOF, that's why the SOF application was passing, but not the kernel tests.
That was actually the problem, that this is not defined in Zephyr also.

**Review by nashif — CHANGES_REQUESTED** [review]
> This symbol is defined in SOF, shouldn't it be configured there instead?

oh, have not noticed that. This cross Kconfig contamination should be avoided. What is CONFIG_IMX exactly? if that is supposed to indicate an SoC family, this needs to come from Zephyr, not SOF. The board, SoC,  architecture kconfigs need to be in Zephyr.

**Review by dbaluta — APPROVED** [review]

**Comment by dbaluta** [comment]
@anasnashif CONIFG_IMX it is supposed to indicate a SoC family.

**Comment by iuliana-prodan** [comment]
> > This symbol is defined in SOF, shouldn't it be configured there instead?
> 
> oh, have not noticed that. This cross Kconfig contamination should be avoided. What is CONFIG_IMX exactly? if that is supposed to indicate an SoC family, this needs to come from Zephyr, not SOF. The board, SoC, architecture kconfigs need to be in Zephyr.

@nashif With this patch, now CONFIG_IMX comes from Zephyr.
If we want to use nxp_adsp_imx8 board it will select the IMX family, for all tests/applications (kernel's or SOF).

We also have a CONFIX_IMX in SOF, and for subsys/audio/sof application this will be selected twice - first from Zephyr and next from SOF.

To fix this I can change the name of the CONFIG_IMX from Zephyr, I can call it CONFIG_SOC_IMX

**Comment by iuliana-prodan** [comment]
> > > This symbol is defined in SOF, shouldn't it be configured there instead?
> > 
> > 
> > oh, have not noticed that. This cross Kconfig contamination should be avoided. What is CONFIG_IMX exactly? if that is supposed to indicate an SoC family, this needs to come from Zephyr, not SOF. The board, SoC, architecture kconfigs need to be in Zephyr.
> 
> @nashif With this patch, now `CONFIG_IMX` comes from Zephyr.
> If we want to use nxp_adsp_imx8 board it will select the IMX family, for all tests/applications (kernel's or SOF).
> 
> We also have a` CONFIX_IMX` in SOF, and for `subsys/audio/sof` application this will be selected twice - first from Zephyr and next from SOF.
> 
> To fix this I can change the name of the `CONFIG_IMX` from Zephyr, I can call it `CONFIG_SOC_IMX`

@nashif @MaureenHelm Actually I've chosen another solution - more generic and it can be used by other platforms too.

So, I've add a generic CONFIG to bypass the size constraint we (and other platforms like Mediatek's [MT8195](https://github.com/thesofproject/sof/blob/main/src/platform/mt8195/include/platform/lib/memory.h#L155)) have for vector table entry - see [commit1](https://github.com/zephyrproject-rtos/zephyr/pull/38431/commits/a9b8d8d4b11216e3e74293ad517bbb8fa2d52cea), and enabled it for `nxp_adsp_imx8` board - see [commit2](https://github.com/zephyrproject-rtos/zephyr/pull/38431/commits/550ff6cd2dee1bf6855fe22974650b1d2d67b94c).

cc: @dbaluta @andyross @dcpleung

**Comment by nashif** [comment]
> @nashif @MaureenHelm Actually I've chosen another solution - more generic and it can be used by other platforms too.

yes, this looks better. We should not have SoC related kconfigs in architecture code anyways. So this is an improvement.

**Review by nashif — APPROVED** [review]

File: arch/xtensa/Kconfig [hunk 1]
```diff
@@ -90,4 +90,13 @@ config XTENSA_WAITI_BUG
 	  platforms which prefixes a WAITI entry with 128 NOP
 	  instructions followed by an ISYNC and EXTW.
 
+config XTENSA_SMALL_VECTOR_TABLE_ENTRY
+	bool "Enable workaround for small vector table entries"
+	default n
```
Comment (MaureenHelm) [inline arch/xtensa/Kconfig:95]: You can remove this line, bools are implicitly default n.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
One minor nit, but otherwise I agree with @nashif this is a better solution.

File: arch/xtensa/Kconfig (hunk 1 above)
Comment (iuliana-prodan) [reply arch/xtensa/Kconfig]: Done!

**Review by MaureenHelm — APPROVED** [review]

File: arch/xtensa/Kconfig (hunk 1 above)
Comment (MaureenHelm) [reply arch/xtensa/Kconfig]: Thank you!

**Review by dleach02 — APPROVED** [review]

**Pull request merged**